python/cudf/dataframe/columnops.py

```diff
@@ -494,7 +500,7 @@ def as_column(arbitrary, nan_as_null=True, dtype=None):
                         np.array(arbitrary, dtype=np_type),
                         nan_as_null=nan_as_null
                     )
-
+    data.name = name
```

Comment:
You should guard against `data.name` being defined and `name` being None here.

Reply:
Yep, have a check now.